### PR TITLE
Migrate to null safety

### PR DESCRIPTION
**Pull request by MichaelMarner**
This PR migrates Redux Remote Devtools to null safety. I'd appreciate feedback from anybody using this, in particular any issues. I've migrated the two example projects and things seem to work without really any modification, but real projects are always different.

Right now this is not publishable, as we will need updates to https://github.com/pauldemarco/socketcluster_client/pull/14 and https://github.com/brianegan/redux_dev_tools/pull/6 to be published to pub.dev.

In the interim I've pointed this project to use my forks of both those packages.

As you can see from the changes, migrating this library to null safety actually required fairly minimal changes. Most of the work was testing due to the [fairly extensive issues](https://github.com/dart-lang/mockito/blob/master/NULL_SAFETY_README.md) Mockito has with null safety.

Counter example migrated with minimal changes. The Github example needed a bit more work, mostly because the code is just so old.

Anyway please point your null safe app at this and tell me what you think

Cheers

### CODE REVIEW

**Comment by nickmeinhold** [comment]
@MichaelMarner thanks for your efforts in chasing down the dependencies and migrating them... you're a legend!  I'll update and let you know if I come across any issues... thanks again 👍

**Comment by nickmeinhold** [comment]
Actually I just noticed in your PR on redux_dev_tools the author said he was going to make some additional changes then publish, maybe it's worth waiting for that before trying out this version?  I imagine not much will change but it sounds like it shouldn't be long to wait... what do you think?

**Comment by MichaelMarner** [comment]
Hey @nickmeinhold - Yep, once the the deps have been published on pub.dev I'll update this PR to point to them and fix up anything that needs it.

Just wanted to get this up for feedback so people can try it out. We're a fair way away from being able to move to sound null safety, but migrating the packages we control is the obvious first step.

File: lib/src/remote_devtools_middleware.dart [hunk 1]
```diff
@@ -27,36 +27,35 @@ class RemoteDevToolsMiddleware<State> extends MiddlewareClass<State> {
...
-  Store<State> store;
-  String _channel;
+  late SocketClusterWrapper socket;
+  Store<State>? store;
+  late String _channel;
   RemoteDevToolsStatus status = RemoteDevToolsStatus.notConnected;
 
   /// The function used to decode actions. If not specifies, defaults to [NopActionDecoder]
-  ActionDecoder actionDecoder;
+  late ActionDecoder /*late*/ actionDecoder;
```
Comment (nickmeinhold) [inline lib/src/remote_devtools_middleware.dart:36]: I assume the /*late*/ is a Hint marker that didn't get updated?  (I saw this earlier and was going to comment after I'd experimented with the PR but missed my chance :-)

Comment (MichaelMarner) [reply lib/src/remote_devtools_middleware.dart:36]: Ah oops, yes. The migration tool _really didn't_ wants respect the hint so I did that migration manually. I'll remove it.

**Pull request merged**